pkg/common/hashmap/joinmap.go

```diff
@@ -21,15 +21,16 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/pb/plan"
 )
 
-func NewJoinMap(sels [][]int64, expr *plan.Expr, mp *StrHashMap, hasNull bool, idx *index.LowCardinalityIndex) *JoinMap {
+func NewJoinMap(sels [][]int64, nullSels []int64, expr *plan.Expr, mp *StrHashMap, hasNull bool, idx *index.LowCardinalityIndex) *JoinMap {
 	cnt := int64(1)
 	return &JoinMap{
-		cnt:     &cnt,
-		mp:      mp,
-		expr:    expr,
-		sels:    sels,
-		hasNull: hasNull,
-		idx:     idx,
+		cnt:      &cnt,
```

Comment:
why you need a &cnt?   A value is not good enough?

Reply:
this field is for reference count of joinmap which is shared by some scopes.   this field had been wrote by  @nnsgmsone   for about 60 days